features/mbedtls/targets/TARGET_CRYPTOCELL310/aes_alt.c

```diff
@@ -861,12 +888,11 @@ int mbedtls_aes_setkey_dec( mbedtls_aes_context *ctx, const unsigned char *key,
         return -1;
 
     if (keybits/8 > SASI_AES_KEY_MAX_SIZE_IN_BYTES)
-        return mbedtls_aes_setkey_sw_dec(&ctx->sw_context,key,keybits);
+        return mbedtls_aes_setkey_sw_dec(ctx,key,keybits);
 
-    ctx->cc_aes_context.CC_cipherFlag = SASI_AES_DECRYPT;
-    ctx->cc_aes_context.CC_KeyData.pKey = ( unsigned char* ) key;
-    ctx->cc_aes_context.CC_KeyData.keySize = keybits / 8;
-    ctx->sw_context.sw_key_set = 0;
+    ctx->rk = ( uint32_t* ) key;
```

Comment:
It will affect performance, but I suppose in this case is security/usability vs performance. Dont think you need to introduce a new variable, we already have a `ctx->buf` which is used for the exact same purpose in the software implementation.